src/main/java/teammates/ui/constants/QuestionTypeStructures.java

```diff
@@ -31,6 +31,7 @@
 public enum QuestionTypeStructures {
     // CHECKSTYLE.OFF:JavadocVariable
     DEFAULT_CONTRIBUTION_QUESTION_DETAILS(new FeedbackContributionQuestionDetails("")),
+    DEFAULT_NEW_CONTRIBUTION_QUESTION_DETAILS(new FeedbackContributionQuestionDetails("", true)),
```

Comment:
Can't really say I like this solution. When are the two default structures used differently? If only one is used, there shouldn't be two of this; this is not likely the right way to achieve backward compatibility either.

Reply:
To add on, default should be isZeroSum = true, notSureAllowed = false.

For the backwards compatibility issue, one possible way would be to look into how the question details are deserialized. Older data do not have the isZeroSum field, so it must have been set to true somewhere.

I'm guessing simply changing the default constructor to something along the lines of this might do the trick. Doing a quick search, this constructor seems to only be used by tests & gson library to deserialize the question details. Might be a possible solution I think is worth considering.
```java
    public FeedbackContributionQuestionDetails() {
        // relevant comment on why this needs to be different
        super(FeedbackQuestionType.CONTRIB, null);
        // set default value of isZeroSum to false here
    }
```

Reply:
@wkurniawan07 `DEFAULT_NEW_CONTRIBUTION_QUESTION_DETAILS()` is used when creating new contribution questions, i.e. in `feedbackQuestionsService.getNewQuestionModel()` while `DEFAULT_CONTRIBUTION_QUESTION_DETAILS()` is used elsewhere so to ensure the value of `isZeroSum` is set to `false` to be compatible with old sessions.

@samuelfangjw ah yes, will change the default values the other way round, i.e. `isZeroSum=true` and `isNotSureAllowed=false`. Following which, `DEFAULT_CONTRIBUTION_QUESTION_DETAILS()` will have the new default values while `DEFAULT_NEW_CONTRIBUTION_QUESTION_DETAILS()` will be renamed to `DEFAULT_OLD_CONTRIBUTION_QUESTION_DETAILS()` and will have the old default values.

Regarding the constructor `FeedbackContributionQuestionDetails()`, are you saying that, instead of creating a new constructor, i.e. `FeedbackContributionQuestionDetails(String questionText, boolean isNewQuestion)`, since this existing constructor is only used in tests, we can modify it to be used to deal with old sessions?

Reply:
Apologies I missed this. I believe the change I suggested is the minimal change needed to maintain backward compatibility. I propose that this be the stopgap measure to maintain compatibility.

Reply:
@samuelfangjw sorry but I'm still a little confused. Does this mean a yes to my previous question?

Reply:
@xyliew25, my answer is irrelevant here, because the question is missing the point entirely. It's not really about modifying unused constructor to deal with old sessions. The point is that this is probably the only point where we are able to differentiate between old and new sessions.

> since this existing constructor is only used in tests

If you read my comment carefully, you will realize that this is not true.

> DEFAULT_NEW_CONTRIBUTION_QUESTION_DETAILS() is used when creating new contribution questions, i.e. in feedbackQuestionsService.getNewQuestionModel() while DEFAULT_CONTRIBUTION_QUESTION_DETAILS() is used elsewhere so to ensure the value of isZeroSum is set to false to be compatible with old sessions.

Please try out my suggested solution, you will realize that none of this is necessary. In fact, I don't believe we can differentiate between "old" and "new" data after the deserialization step because old and new data have the same properties after this point. To elaborate, the only way we can differentiate "old" and "new" data is the presence/absence of isZeroSum property. After it is deserialized, isZeroSum is set to either true or false and so becomes indistinguishable from "new" data. 

I'm not entirely sure how you are differentiating between old and new data, but if you are relying on isZeroSum to not be set after this point it won't work. This is a question using old data using your current implementation:

<img width="575" alt="Screen Shot 2022-06-26 at 4 10 25 PM" src="https://user-images.githubusercontent.com/60355570/175805557-b91b5b83-8b00-4750-a4de-c565d8fe7715.png">

Reply:
I understand that the solution above has a few logical gaps that might be hard to understand. Wasn't meant as a complete solution, but to help guide this PR towards a possible feasible solution. Following the same line of thought, we could also do this:

1. Use `Boolean` wrapper class instead of `boolean` for `isZeroSum` and ensure this property is not set in the constructor used for deserialization.
2. By default, if property is absent (old data) `isZeroSum` will be `null` instead of a boolean value.
3. After deserialization, we can now use presence or absence of this property to determine whether this is old or new data (and hence if `isZeroSum` should be true or false).

A bit more involved than the suggested one above, but may be a bit cleaner because we do not make the assumption that the default constructor is only used for deserialization in production code. 

Regardless of what solution is chose, I believe any feasible solution must involve the point between when the data is fetched and when the string is deserialized, because with the current implementation it's simply not possible to distinguish older data from newer data past this point. Do let me know if i've missed something.

Reply:
@samuelfangjw thank you for the detailed elaboration! Will look into it and update you again.

Reply:
@samuelfangjw I have updated the code. Would appreciate it if you could review it again.